apps/frontend/src/app/5_pages/LendPage/components/LendModal/LendingForm.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+import { Trans } from 'react-i18next';
+
+import {
+  AmountInput,
+  Button,
+  Checkbox,
+  FormGroup,
+  SimpleTable,
+  SimpleTableRow,
+  Link,
+  ErrorBadge,
+  ErrorLevel,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { defaultChainId } from '../../../../../config/chains';
+
+import { AmountRenderer } from '../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AssetRenderer } from '../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { MaxButton } from '../../../../2_molecules/MaxButton/MaxButton';
+import { GAS_LIMIT } from '../../../../../constants/gasLimits';
+import { useMaintenance } from '../../../../../hooks/useMaintenance';
+import { useMaxAssetBalance } from '../../../../../hooks/useMaxAssetBalance';
+import { useWeiAmountInput } from '../../../../../hooks/useWeiAmountInput';
+import { translations } from '../../../../../locales/i18n';
+import { asyncCall } from '../../../../../store/rxjs/provider-cache';
+import { FullLendingModalState } from './LendingModalContainer';
+
+export type DepositProps = {
+  state: FullLendingModalState;
+  onConfirm: (amount: Decimal) => void;
+};
+
+export const LendingForm: FC<DepositProps> = ({ state, onConfirm }) => {
+  const [value, setValue, amount] = useWeiAmountInput('');
+
+  const { checkMaintenance, States } = useMaintenance();
+  const depositLocked = checkMaintenance(States.DEPOSIT_LEND);
+
+  const { balance } = useMaxAssetBalance(
+    state.token,
+    defaultChainId,
+    GAS_LIMIT.LENDING_MINT,
+  );
+
+  const [hasDisclaimerBeenChecked, setHasDisclaimerBeenChecked] =
+    useState(false);
+
+  const handleSubmit = useCallback(
+    () => onConfirm(Decimal.fromBigNumberString(amount.toString())),
+    [amount, onConfirm],
+  );
+
+  const handleMaxClick = useCallback(
+    () => setValue(balance.toString()),
+    [balance, setValue],
+  );
+
+  const isSubmitDisabled = useMemo(
+    () =>
+      amount.lte(0) ||
+      amount.gt(balance.toBigNumber()) ||
+      !hasDisclaimerBeenChecked ||
+      depositLocked,
+    [amount, balance, hasDisclaimerBeenChecked, depositLocked],
+  );
+
+  const [lendApy, setLendApy] = useState<Decimal>(state.apy);
+
+  useEffect(() => {
+    asyncCall(
+      `lending/${state.poolTokenContract.address}/nextSupplyInterestRate/${amount}`,
+      () => state.poolTokenContract.nextSupplyInterestRate(amount),
```

Comment:
after changing the destination branch of this PR to [`SOV-2738: Develop D2 Lend page`] PR you can just use a hook from `apps/frontend/src/app/5_pages/LendPage/hooks/useGetNextSupplyInterestRate.ts`